### DISCUSSION TITLE
Flow ratio shown in object doesn't match flow ratio configured in filament settings area

### DISCUSSION THREAD

**Discussion by hindsight351** [OrcaSlicer/OrcaSlicer#1120] [Q&A] [answered]
I'm not sure if this is a bug or if I'm doing something wrong or not understanding how this works. But I'm trying to calibrate my flow ratio. If I go in and set the flow ratio to say 0.95 in the Filament settings, then slice a calibration cube, then change to object view in the process settings area for the cube I sliced, under the Quality settings, I see a Flow Ratio of "1". I would expect it to be 0.95 to match what I set in the filaments setting area. 

Am I missing something, doing something wrong? Or does "1" in the object view mean that the flow ratio will be set to exactly what is in the filament setting (0.95 in this case)...... in other words, if I have 0.95 in filament setting area, and 0.9 in Object area, it means that the actual effective flow ratio will be 90% of 0.95?

Edit: I just performed a test where I increased the flow ratio to "1.08" in the filament setting area, sliced, verified that the flow ratio in the Object setting was still "1", and my cube wall size increased. Which suggests that the flow ratio shown in the Object setting is a *modifier* of whatever is in the filament settings area. Can anyone confirm this is the intended behavior?

**Comment by julie777** [comment]
I am not sure of this, but I would expect the filament to be a modifier for the number in the process settings.

You select your settings. That is the ideal, so regardless which filament used, the setting is x, where x in 1 in your case for flow, or it could be default speed of 100.

You select your filament and the slicer uses the process settings modified by the filament settings to do the actual slicing. So in your case the slicer is actually using 1 times 0.95 = 0.95.

If you set the process to use .95 then the slicer would use .95 times .95 which would be too low.

One example of this is the speed can be set to 100 in process settings for the outer perimeter, but if the filament settings have slow down for layers with layer time below the set value, then the preview will show the actual speed that was used. Flow rate works the same way. If the speed setting would cause the volume flow rate to exceed the max specified in the filament settings then the output of slicing will show a lower speed being used.

**Reply by hindsight351** [reply]
Thank you

**Comment by SoftFever** [comment, marked as answer]
@hindsight351 
`Which suggests that the flow ratio shown in the Object setting is a modifier of whatever is in the filament settings area.`
This is correct

**Reply by hindsight351** [reply]
Thank you

**Reply by hindsight351** [reply]
> @hindsight351 `Which suggests that the flow ratio shown in the Object setting is a modifier of whatever is in the filament settings area.` This is correct

What about the top and bottom flow ratio modifiers? In the per-object level settings, I can set modifiers for flow ratio, top flow ratio, and bottom flow ratio. If my main filament-level flow ratio is 1, and I set an object level flow ratio of 0.9 and a bottom flow ratio of 0.9, does that mean my bottom flow ratio will be 0.9 or 0.81?

**Comment by JendaDH** [comment]
Hi, 

so to get it right, if flow ratio is set to 0.9 in the filament settings, then to have the same value on top and bottom surfaces, the value in advanced section should be 1. 

If it is set to anything else, it will either lower or increase the filament flow ratio based on the value (above / below 1) => final flow ratio = advanced object flow parameter * filament flow parameter.

Is my understanding right?

**Reply by SoftFever** [reply]
correct

**Comment by Unhallowed1949** [comment]
To go further into this, if I'm doing a "Pass 2" of the flow claibration, and my filament setting flow ratio is set to 1.05, the per object settings will be as follows cube 0 = 1, cube -1 = 0.99, cube -2 = 0.98, etc, etc. So, if I choose cube -2 as my correct flow ratio, does that mean I take 1.05 x 0.98 = 1.029 or does that mean I take 1.05 - 0.02 = 1.03? I believe it's the former but I would just like to confirm @SoftFever

**Comment by PoiMechanic** [comment]
I am having this issue too. Flow Ratio in Objects keeps resetting to 1 but Top and Bottom Flow Ratio in Objects changes to whatever I have my Filament saved to. Doing flow rate cubes at slow speed I was getting perfect .6 walls and I would set my filament to whatever flow gave me that reading (1.13). But if I sped up from 30mms to 50mms I would get .5 walls on that test. I turned my Max Volumetric Flow up to 30 and nothing changed, still .5 walls. I finally realized it was because Objects Flow Ratio keeps resetting to 1 but if I turn it up to 1.13 then I get .6 walls again. So that means 1 doesn't represent 1 x the Filament Flow Rate but rather flow rate itself. Why should I have to change it every time I start a new project? Doesn't matter if I save it, it always resets.
